sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,154 @@
+                } else if (annotation instanceof SearchableFieldProperty) {
+                    buildSearchableField(searchField, (SearchableFieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                } else if (annotation instanceof FieldProperty) {
+                    buildField(searchField, (FieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                }
+                logger.info("The property {} is not a search field.", field.getName());
+            }
+        }
+        return searchFieldList;
+    }
+
+    private static void buildSimpleField(Field searchField, SimpleFieldProperty annotation) {
+        searchField.setSearchable(true);
+        searchField.setSortable(true);
+        searchField.setFilterable(true);
+        searchField.setFacetable(true);
+        searchField.setKey(annotation.isKey());
+        if (!"null".equals(annotation.analyzer())) {
+            searchField.setAnalyzer(AnalyzerName.fromString((annotation.analyzer())));
+        }
+        if (!"null".equals(annotation.searchAnalyzer())) {
+            searchField.setAnalyzer(AnalyzerName.fromString((annotation.searchAnalyzer())));
+        }
+        if (!"null".equals(annotation.indexAnalyzer())) {
+            searchField.setAnalyzer(AnalyzerName.fromString((annotation.indexAnalyzer())));
+        }
+        if (annotation.synonymMaps().length != 0) {
+            searchField.setSynonymMaps(Arrays.asList(annotation.synonymMaps()));
+        }
+    }
+
+    private static void buildSearchableField(Field searchField, SearchableFieldProperty annotation) {
+        searchField.setSearchable(true);
+        searchField.setSortable(annotation.isSortable());
+        searchField.setFilterable(annotation.isFilterable());
+        searchField.setFacetable(annotation.isFacetable());
+        searchField.setKey(annotation.isKey());
+        if (!"null".equals(annotation.analyzer())) {
+            searchField.setAnalyzer(AnalyzerName.fromString((annotation.analyzer())));
+        }
+        if (!"null".equals(annotation.searchAnalyzer())) {
+            searchField.setAnalyzer(AnalyzerName.fromString((annotation.searchAnalyzer())));
+        }
+        if (!"null".equals(annotation.indexAnalyzer())) {
+            searchField.setAnalyzer(AnalyzerName.fromString((annotation.indexAnalyzer())));
+        }
+        if (annotation.synonymMaps().length != 0) {
+            searchField.setSynonymMaps(Arrays.asList(annotation.synonymMaps()));
+        }
+    }
+
+    private static void buildField(Field searchField, FieldProperty annotation) {
+        searchField.setSearchable(annotation.isSearchable());
+        searchField.setSortable(annotation.isSortable());
+        searchField.setFilterable(annotation.isFilterable());
+        searchField.setFacetable(annotation.isFacetable());
+        searchField.setKey(annotation.isKey());
+        if (!"null".equals(annotation.analyzer())) {
```

Comment:
This should be a string constant "null". But why is the default "null" rather than just a null object?

Reply:
Annotation for string method cannot default to null object. The field is not required, so I made it default to null string.